internal/primitive/transform/action/structs/duplicate.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Linkall Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package structs
+
+import (
+	"fmt"
+
+	"github.com/linkall-labs/vanus/internal/primitive/transform/action"
+	"github.com/linkall-labs/vanus/internal/primitive/transform/arg"
+	"github.com/linkall-labs/vanus/internal/primitive/transform/common"
+	"github.com/linkall-labs/vanus/internal/primitive/transform/context"
+)
+
+type duplicateAction struct {
+	action.CommonAction
+}
+
+// NewDuplicateAction ["duplicate", "sourcePath", "targetPath"].
+func NewDuplicateAction() action.Action {
+	return &duplicateAction{
+		action.CommonAction{
+			ActionName: "DUPLICATE",
+			FixedArgs:  []arg.TypeList{arg.EventList, arg.All},
```

Comment:
```suggestion
			FixedArgs:  []arg.TypeList{arg.EventList, arg.EventList},
```

Reply:
@n-kurasawa The second arg as a "targetPath" can be either an attribute path (like `$.newKey`) or a data path (like `$.data.newKey`). So we should use arg.EventList to limit the type. `arg.All` includes constant which is not applicable for this function.
```
var (
   EventList = []Type{EventAttribute, EventData}
   All       = []Type{EventAttribute, EventData, Constant, Define}
)
```

Reply:
Thanks for the explanation.
fixed it.

Reply:
This PR count 5 contribution points for our [Top-10 Active contributors Event](https://www.vanus.dev/community/community-events/top-10-active-contributors-2022). Congrats @n-kurasawa , you're already in Top 10 contributors list ! Join the [Slack channel](https://join.slack.com/t/vanusworkspace/shared_invite/zt-1jilbbfo2-NxiFG0VOo8ABGCCNaeNfcA) to discuss with other developers and know details about how to get event rewards.